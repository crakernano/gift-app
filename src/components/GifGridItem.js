import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = (img)=>{
    return(
        <div className="card animate_fadeIn">
            <img src={img.url} alt={img.title} />
            <p>{img.title}</p>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}