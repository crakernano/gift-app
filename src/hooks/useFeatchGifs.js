import React, {useState, useEffect} from 'react'
import { getGif } from "../helpers/getGifs";

export const useFeatchGifs = ( categoria ) => {
    const [state, setState] = useState({
        data : [],
        loading: true
    });

    useEffect( () => {
        getGif( categoria )
        .then(img => {setState({
            data: img,
            loading: false
        });
    })

    }, [ categoria ])

    return state;
}