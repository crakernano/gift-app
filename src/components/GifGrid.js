import React from 'react'
import { useFeatchGifs } from '../hooks/useFeatchGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';

export const GifGrid = ({category}) => {

//    const [images, setImages] = useState([]);


    const {data, loading} = useFeatchGifs( category );

    return (
        <div className="card-grid">
            <h3>{category}</h3>
            {loading ? 'Cargando...' : ''}
                {
                data.map(img => (
                    <GifGridItem 
                        key={img.id}
                        { ...img }
                    />
                ))
                }
        </div>
    )
}

GifGrid.protoType = {
    category: PropTypes.string.isRequired
}