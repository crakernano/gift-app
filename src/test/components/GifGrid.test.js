import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFeatchGifs } from '../../hooks/useFeatchGifs';
jest.mock('../../hooks/useFeatchGifs');

describe('Pruebas en <GifGrid />', ()=>{

    const category = "Pokemon";
    

    test('Debe mostrarse correctamente', ()=>{


        useFeatchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();
    })

    test('Debe mostrar los items cuadno userFeatchGif devuelve datos', ()=>{
        
        const imgs = [{
            id: 'ABC',
            url: 'https://localhost/imagen.png',
            title: 'Titulo'
        }];
        
        useFeatchGifs.mockReturnValue({
            data: imgs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGridItem').length).toBe(imgs.length);
    })
})