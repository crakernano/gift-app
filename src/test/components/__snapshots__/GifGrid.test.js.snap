// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> Debe mostrar los items cuadno userFeatchGif devuelve datos 1`] = `
<div
  className="card-grid"
>
  <h3>
    Pokemon
  </h3>
  <GifGridItem
    id="ABC"
    key="ABC"
    title="Titulo"
    url="https://localhost/imagen.png"
  />
</div>
`;

exports[`Pruebas en <GifGrid /> Debe mostrarse correctamente 1`] = `
<div
  className="card-grid"
>
  <h3>
    Pokemon
  </h3>
  Cargando...
</div>
`;
